source/develop/release-management/features/virt/vm-lifecycle-with-kubevirt.md

```diff
@@ -0,0 +1,130 @@
+
+Then, the mentioned run-VM flow will be modified. The only change would be that in Kubvirt-cluster, CreateBrokerVDSCommand will generate a bit different configuration of the VM and will send it to Kubevirt (using the Kubernetes API) rather than VDSM. Note that the scheduling is still done in oVirt.  
+
+The represenation of the VM will be as follow (in JSON):  
+1. Domxml (Kubevirt passes it as-is to the node at this stage, later on Kubervirt will replace placeholders within the XML).  
+2. The scheduled host (Kubvirt uses this as the destination node).
+
+### Benefits
+1. First notion of Kubernetes in oVirt + start getting feedback on running real-world virtual machines with Kubevirt.    
+2. We bypass the 'create' verb in the engine<->vdsm API.  
+3. We do not depend on having a full blown scheduler in Kubervirt.
+
+### Limitations & Challenges
+1. At this stage the engine does not do the operations that it used to do after the detination host was chosen and before sending the request to VDSM.  
+2. The engine cannot lock the monitoring until VDSM acknowlege that the request arrives. So we need to set a timeout so if the VM is still down afterwards, we'll conclude that the run operation failed.
+3. oVirt still schedules the nodes.
+4. oVirt still monitors the nodes as it used before.  
+5. Most of the VM related operations are still done directly with VDSM (rather than through Kubervirt).
+
+## Milestone 2: Scheduling
+
+### Goal
+Basic VM schedluing in Kubevirt.
+
+### Design
+1. Extend the VM representation with cluster-related data, e.g., host-pinning. This will be represented as properties in JSON. VM-related data needs to be retrieved from the VM configuration (see VM Represenation section).  
+2. Adjust VMs monitoring as described above. That is, to add a timeout for run-vm so the monitoring will end the run-vm operation if the VM is down after some time.  
+3. Some basic mointoring should be done in Kubevirt in order to rerun a virtual machine (oVirt cannot do this at this phase since the engine doesn't know which nodes were previously chosen). Kubevirt is reponsible to detecting failures to run the VM on the node and to attempt to reschedule the VM on another node.
+
+### Benefits
+1. We start to extract the scheduling part from oVirt.  
+2. We add basic monitoring capability to Kubevirt (to examine the state of the VM on the scheduled node and retrieve information that is needed for scheduling).  
+3. We extract the rerun mechanism for run-VM from oVirt.  
+
+### Limitations & Challenges
+1. Either use the 'stop-the-world' approach of the current scheduler in oVirt or invent an alternative approach.  
+2. The scheduling does not take into consideration host-devices.  
```

Comment:
So we have to assume that we have an impact on where the pod runs because virt-handler is supposed to run the VM on the node that pod was scheduled to, right?
As long as either the scheduling or the monitoring is done by oVirt, I imagine that each node has a label with its identifier in oVirt so once the nodeName field of the VM objects is set to that identifier the VM specification is magically reaches the node and virt-handler starts the VM